linalg/densemat.hpp

```diff
@@ -355,6 +355,51 @@ class DenseMatrix : public Matrix
    /// Perform (ro+i,co+j)+=a*A(i,j) for 0<=i<A.Height, 0<=j<A.Width
    void AddMatrix(double a, const DenseMatrix &A, int ro, int co);
 
+   /// Get the square submatrix which corresponds to the given indices @a idx
```

Comment:
In the documentation for these `GetSubMatrix` methods we should add note that the `A` matrix will be resized to accommodate the data.

Reply:
done